File: schemas/booking-scheduling/drizzle/schedule_rules.ts
// schedule_rules: recurring weekly availability windows within a schedule.
// See README.md for full design rationale.

import { pgTable, uuid, integer, text, timestamp, index } from "drizzle-orm/pg-core";
import { schedules } from "./schedules";

export const scheduleRules = pgTable(
  "schedule_rules",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    scheduleId: uuid("schedule_id").notNull().references(() => schedules.id, { onDelete: "cascade" }),
    dayOfWeek: integer("day_of_week").notNull(),
    startTime: text("start_time").notNull(),
    endTime: text("end_time").notNull(),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_schedule_rules_schedule_id_day_of_week").on(table.scheduleId, table.dayOfWeek),
  ]
);

File: schemas/booking-scheduling/spacetimedb/schedule_rules.rs
// schedule_rules: Recurring weekly availability rules within a schedule.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = schedule_rules, public)]
pub struct ScheduleRule {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub schedule_id: String, // UUID — FK → schedules.id (cascade delete)
    pub day_of_week: i32,
    pub start_time: String,
    pub end_time: String,
    pub created_at: Timestamp,
    // Composite index: (schedule_id, day_of_week)
}
